File: app/src/main/java/com/communityx/fragments/SignUpRoleFragment.java
package com.communityx.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import com.communityx.R;
import com.communityx.R.id;
import com.communityx.activity.SignUpStudentInfoActivity;

import java.util.Objects;

public class SignUpRoleFragment extends Fragment {

    @BindView(id.view_freshman_main)
    RelativeLayout viewFreshman;
    @BindView(id.view_sophomore_main)
    RelativeLayout viewSophomore;
    @BindView(id.view_junior_main)
    RelativeLayout viewJunior;
    @BindView(id.view_senior_main)
    RelativeLayout viewSenior;
    @BindView(id.image_freshman)
    ImageView imageFreshman;
    @BindView(id.image_sophomore)
    ImageView imageSophomore;
    @BindView(id.image_junior)
    ImageView imageJunior;
    @BindView(id.image_senior)
    ImageView imageSenior;
    @BindView(id.image_freshman_tick)
    ImageView tickFreshman;
    @BindView(id.image_sophomore_tick)
    ImageView tickSophomore;
    @BindView(id.image_junior_tick)
    ImageView tickJunior;
    @BindView(id.image_senior_tick)
    ImageView tickSenior;
    @BindView(id.text_freshman)
    TextView textFreshman;
    @BindView(id.text_sophomore)
    TextView textSophomore;
    @BindView(id.text_junior)
    TextView textJunior;
    @BindView(id.text_senior)
    TextView textSenior;

    private enum Role{
        FRESHMAN,
        SOPHOMORE,
        JUNIOR,
        SENIOR
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_sign_up_select_role,null);
        ButterKnife.bind(this,view);
        return view;
    }

    @OnClick({id.view_freshman_main,id.view_sophomore_main,id.view_junior_main,id.view_senior_main})
    void tappedRole(View view){
        if (view.equals(viewFreshman)) selectRole(Role.FRESHMAN);
        else if (view.equals(viewSophomore)) selectRole(Role.SOPHOMORE);
        else if (view.equals(viewJunior)) selectRole(Role.JUNIOR);
        else if (view.equals(viewSenior)) selectRole(Role.SENIOR);
    }

    private void selectRole(Role freshman) {
        ((SignUpStudentInfoActivity) Objects.requireNonNull(getActivity())).buttonContinue.setBackground(this.getResources().getDrawable(R.drawable.button_active));
        ((SignUpStudentInfoActivity) Objects.requireNonNull(getActivity())).buttonContinue.setTag(true);

        switch (freshman){
            case FRESHMAN:
                viewFreshman.setBackground(getActivity().getResources().getDrawable(R.drawable.border_orange_bg));
                viewSophomore.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewJunior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewSenior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));

                imageFreshman.setImageResource(R.drawable.ic_signup_freshman_select);
                imageSophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect);
                imageJunior.setImageResource(R.drawable.ic_signup_junior_deselect);
                imageSenior.setImageResource(R.drawable.ic_signup_senior_deselect);

                textFreshman.setTextColor(this.getResources().getColor(R.color.colorBlackTitle));
                textSophomore.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textJunior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textSenior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));

                tickFreshman.setVisibility(View.VISIBLE);
                tickSophomore.setVisibility(View.GONE);
                tickJunior.setVisibility(View.GONE);
                tickSenior.setVisibility(View.GONE);
                break;

            case SOPHOMORE:
                viewSophomore.setBackground(getActivity().getResources().getDrawable(R.drawable.border_orange_bg));
                viewFreshman.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewJunior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewSenior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));

                imageSophomore.setImageResource(R.drawable.ic_signup_sophomore_select);
                imageFreshman.setImageResource(R.drawable.ic_signup_freshman_deselect);
                imageJunior.setImageResource(R.drawable.ic_signup_junior_deselect);
                imageSenior.setImageResource(R.drawable.ic_signup_senior_deselect);

                textSophomore.setTextColor(this.getResources().getColor(R.color.colorBlackTitle));
                textFreshman.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textJunior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textSenior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));

                tickSophomore.setVisibility(View.VISIBLE);
                tickFreshman.setVisibility(View.GONE);
                tickJunior.setVisibility(View.GONE);
                tickSenior.setVisibility(View.GONE);
                break;

            case JUNIOR:
                viewJunior.setBackground(getActivity().getResources().getDrawable(R.drawable.border_orange_bg));
                viewFreshman.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewSophomore.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewSenior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));

                imageJunior.setImageResource(R.drawable.ic_signup_junior_select);
                imageFreshman.setImageResource(R.drawable.ic_signup_freshman_deselect);
                imageSophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect);
                imageSenior.setImageResource(R.drawable.ic_signup_senior_deselect);

                textJunior.setTextColor(this.getResources().getColor(R.color.colorBlackTitle));
                textFreshman.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textSophomore.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textSenior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));

                tickJunior.setVisibility(View.VISIBLE);
                tickFreshman.setVisibility(View.GONE);
                tickSophomore.setVisibility(View.GONE);
                tickSenior.setVisibility(View.GONE);
                break;

            case SENIOR:
                viewSenior.setBackground(getActivity().getResources().getDrawable(R.drawable.border_orange_bg));
                viewFreshman.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewSophomore.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));
                viewJunior.setBackground(getActivity().getResources().getDrawable(R.drawable.bordered_bg));

                imageSenior.setImageResource(R.drawable.ic_signup_senior_select);
                imageFreshman.setImageResource(R.drawable.ic_signup_freshman_deselect);
                imageSophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect);
                imageJunior.setImageResource(R.drawable.ic_signup_junior_deselect);

                textSenior.setTextColor(this.getResources().getColor(R.color.colorBlackTitle));
                textFreshman.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textSophomore.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));
                textJunior.setTextColor(this.getResources().getColor(R.color.colorLightestGrey));

                tickSenior.setVisibility(View.VISIBLE);
                tickFreshman.setVisibility(View.GONE);
                tickSophomore.setVisibility(View.GONE);
                tickJunior.setVisibility(View.GONE);
                break;
        }
    }
}

File: app/src/main/java/com/communityx/fragments/SignUpRoleFragment.kt
package com.communityx.fragments

import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.communityx.R
import com.communityx.activity.SignUpStudentInfoActivity
import com.communityx.base.BaseSignUpFragment
import com.communityx.models.signup.DataX
import com.communityx.models.signup.SignUpRequest
import com.communityx.models.signup.club.ClubResponse
import com.communityx.models.signup.institute.InstituteRequest
import com.communityx.network.ResponseListener
import com.communityx.network.serviceRepo.SignUpRepo
import com.communityx.utils.*
import com.communityx.utils.AppConstant.PREF_USER_ID
import com.communityx.utils.AppConstant.USER_ID
import kotlinx.android.synthetic.main.fragment_sign_up_select_role.*
import java.util.*

class SignUpRoleFragment : BaseSignUpFragment(), View.OnClickListener , AppConstant {

    private var standardYear: String? =null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_sign_up_select_role, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        initField()
        enableButton(true)
        view_freshman_main.setOnClickListener(this)
        view_sophomore_main.setOnClickListener(this)
        view_junior_main.setOnClickListener(this)
        view_senior_main.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        tappedRole(v)
    }

    override fun onContinueButtonClicked() {
        if(setFieldsData()) {
            addInstitute()
        }
    }

    override fun setFieldsData(): Boolean {
        signUpStudent?.standard_year = standardYear
        return validateEmpty(signUpStudent)
    }

    override fun validateEmpty(requestData: SignUpRequest?, showSnackbar: Boolean): Boolean {
        val b = !TextUtils.isEmpty(requestData?.standard_year)
        if (!b && showSnackbar) SnackBarFactory.createSnackBar(
            context,
            constraint_layout,
            getString(R.string.please_select_qualification)
        )
        return  b
    }

    private fun initField() {
        if(validateEmpty(signUpStudent,false)) {
            selectRole(Role.valueOf(signUpStudent?.standard_year!!))
            enableButton(true)
        }
    }

    private fun tappedRole(view: View?) {
        when (view) {
            view_freshman_main -> selectRole(Role.FRESHMAN)
            view_sophomore_main -> selectRole(Role.SOPHOMORE)
            view_junior_main -> selectRole(Role.JUNIOR)
            view_senior_main -> selectRole(Role.SENIOR)
        }
    }

    private fun selectRole(role: Role) {
        (Objects.requireNonNull<FragmentActivity>(activity) as SignUpStudentInfoActivity).enableButton(true)
        standardYear = role.name
        when (role) {
            SignUpRoleFragment.Role.FRESHMAN -> {
                view_freshman_main.background = resources.getDrawable(R.drawable.border_orange_bg)
                view_sophomore_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_junior_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_senior_main.background = resources.getDrawable(R.drawable.bordered_bg)

                image_freshman.setImageResource(R.drawable.ic_signup_freshman_select)
                image_sophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect)
                image_junior.setImageResource(R.drawable.ic_signup_junior_deselect)
                image_senior.setImageResource(R.drawable.ic_signup_senior_deselect)

                text_freshman.setTextColor(this.resources.getColor(R.color.colorBlackTitle))
                text_sophomore.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_junior.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_senior.setTextColor(this.resources.getColor(R.color.colorLightestGrey))

                image_freshman_tick.visibility = View.VISIBLE
                image_sophomore_tick.visibility = View.GONE
                image_junior_tick.visibility = View.GONE
                image_senior_tick.visibility = View.GONE
            }

            SignUpRoleFragment.Role.SOPHOMORE -> {
                view_sophomore_main.background = resources.getDrawable(R.drawable.border_orange_bg)
                view_freshman_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_junior_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_senior_main.background = resources.getDrawable(R.drawable.bordered_bg)

                image_sophomore.setImageResource(R.drawable.ic_signup_sophomore_select)
                image_freshman.setImageResource(R.drawable.ic_signup_freshman_deselect)
                image_junior.setImageResource(R.drawable.ic_signup_junior_deselect)
                image_senior.setImageResource(R.drawable.ic_signup_senior_deselect)

                text_sophomore.setTextColor(this.resources.getColor(R.color.colorBlackTitle))
                text_freshman.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_junior.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_senior.setTextColor(this.resources.getColor(R.color.colorLightestGrey))

                image_sophomore_tick.visibility = View.VISIBLE
                image_freshman_tick.visibility = View.GONE
                image_junior_tick.visibility = View.GONE
                image_senior_tick.visibility = View.GONE
            }

            SignUpRoleFragment.Role.JUNIOR -> {
                view_junior_main.background = resources.getDrawable(R.drawable.border_orange_bg)
                view_freshman_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_sophomore_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_senior_main.background = resources.getDrawable(R.drawable.bordered_bg)

                image_junior.setImageResource(R.drawable.ic_signup_junior_select)
                image_freshman.setImageResource(R.drawable.ic_signup_freshman_deselect)
                image_sophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect)
                image_senior.setImageResource(R.drawable.ic_signup_senior_deselect)

                text_junior.setTextColor(resources.getColor(R.color.colorBlackTitle))
                text_freshman.setTextColor(resources.getColor(R.color.colorLightestGrey))
                text_sophomore.setTextColor(resources.getColor(R.color.colorLightestGrey))
                text_senior.setTextColor(resources.getColor(R.color.colorLightestGrey))

                image_junior_tick.visibility = View.VISIBLE
                image_freshman_tick.visibility = View.GONE
                image_sophomore_tick.visibility = View.GONE
                image_senior_tick.visibility = View.GONE
            }

            SignUpRoleFragment.Role.SENIOR -> {
                view_senior_main.background = resources.getDrawable(R.drawable.border_orange_bg)
                view_freshman_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_sophomore_main.background = resources.getDrawable(R.drawable.bordered_bg)
                view_junior_main.background = resources.getDrawable(R.drawable.bordered_bg)

                image_senior.setImageResource(R.drawable.ic_signup_senior_select)
                image_freshman.setImageResource(R.drawable.ic_signup_freshman_deselect)
                image_sophomore.setImageResource(R.drawable.ic_signup_sophomore_deselect)
                image_junior.setImageResource(R.drawable.ic_signup_junior_deselect)

                text_senior.setTextColor(this.resources.getColor(R.color.colorBlackTitle))
                text_freshman.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_sophomore.setTextColor(this.resources.getColor(R.color.colorLightestGrey))
                text_junior.setTextColor(this.resources.getColor(R.color.colorLightestGrey))

                image_senior_tick.visibility = View.VISIBLE
                image_freshman_tick.visibility = View.GONE
                image_sophomore_tick.visibility = View.GONE
                image_junior_tick.visibility = View.GONE
            }
        }
    }

    private fun addInstitute() {
        val instituteRequest = InstituteRequest(
            signUpStudent?.standard_name!!,
            signUpStudent?.standard_year!!,
            signUpStudent?.standard!!,
            AppPreference.getInstance(activity!!).getString(PREF_USER_ID))

        var dialog = CustomProgressBar.getInstance(activity!!).showProgressDialog(getString(R.string.please_wait_while_register))
        SignUpRepo.addInstitute(activity!!, instituteRequest, object : ResponseListener<List<DataX>> {
            override fun onSuccess(response: List<DataX>) {
                dialog.dismiss()
                changeButtonStatus(2, true)
                goToNextPage()
                enableButton(true)
            }

            override fun onError(error: Any) {
                Utils.showError(activity, constraint_layout, error)
                dialog.dismiss()
            }
        })
    }

    private enum class Role {
        FRESHMAN,
        SOPHOMORE,
        JUNIOR,
        SENIOR
    }
}
